Added exogenous variables for better / more accurate forecasting (#31)
ServiceNow/stl-decomp-4j: README.md

Limited testing show the Java to be about half the speed of the Fortran. A comparison of the resulting decompositions is shown in the [StlJavaFortranComparison notebook](examples/StlPerfTest/StlJavaFortranComparison.ipynb).

## Extension to include exogenous inputs 

The STL method is extended to incorporate exogenous inputs where details can be found in the [StlExogenous example](https://github.com/ServiceNow/stl-decomp-4j/tree/master/examples/StlExogenous). 

> To apply these inputs, simply build the smoother with them (double[][] exogenousinputs with each row being an input) as illustrated below. Afterwards, the .decompose() method is again applied and the .getTrend() now extracts the combined effect of the trend and exogenous inputs. 

Reviewer: Small cosmetic suggestion: please wrap the method and function references in single quotes (`) for better readability.

> To apply these inputs, simply build the smoother with them (double[][] exogenousinputs with each row being an input) as illustrated below. Afterwards, the .decompose() method is again applied and the .getTrend() now extracts the combined effect of the trend and exogenous inputs. 

Reviewer: ```suggestion
To apply these inputs, simply build the smoother with them (`double[][] exogenousinputs` with each row being an input) as illustrated below. Afterwards, the `.decompose()` method is again applied and the `.getTrend()` now extracts the combined effect of the trend and exogenous inputs. 
```

> To apply these inputs, simply build the smoother with them (`double[][] exogenousinputs` with each row being an input) as illustrated below. Afterwards, the `.decompose()` method is again applied and the `.getTrend()` now extracts the combined effect of the trend and exogenous inputs. 

Reviewer: ```suggestion
To apply exogenous inputs, build the smoother with them (`double[][] exogenousinputs` with each row being one exogenous input) as shown in the example below. After calling the `.decompose()` method, you can get the STL components. Note that the `.getTrend()` method now extracts the combined effect of the trend and exogenous inputs. 
```

```java 

> SeasonalTrendLoess smoother = builder.buildSmoother(values, exogenousinputs) 

Reviewer: ```suggestion
SeasonalTrendLoess smoother = builder.buildSmoother(values, exogenousinputs);
```
